src/plotSearch/components/directReservationLinkModal/DirectReservationLinkModal.js

```diff
@@ -0,0 +1,289 @@
+// @flow
+import React, {Component, type Element, Fragment} from 'react';
+import {flowRight} from 'lodash/util';
+import {connect} from 'react-redux';
+import {FieldArray, getFormValues, reduxForm, change, initialize} from 'redux-form';
+import {Row, Column} from 'react-foundation';
+
+import Modal from '$components/modal/Modal';
+import ModalButtonWrapper from '$components/modal/ModalButtonWrapper';
+import Button from '$components/button/Button';
+import {ButtonColors} from '$components/enums';
+import {FormNames} from '$src/enums';
+import FormField from '../../../components/form/FormField';
+import {FieldTypes} from '../../../enums';
+import FormFieldLabel from '../../../components/form/FormFieldLabel';
+import LoaderWrapper from '../../../components/loader/LoaderWrapper';
+import Loader from '../../../components/loader/Loader';
+import {getIsCreatingDirectReservationLink} from '../../../plotSearch/selectors';
+import {createDirectReservationLink} from '../../../plotSearch/actions';
+
+type OwnProps = {
+  isOpen: boolean,
+  onClose: () => void,
+  targets: Object[] | null,
+};
+
+type Props = {
+  ...OwnProps,
+  change: Function,
+  initialize: Function,
+  reset: () => void,
+  formValues: Object,
+  isCreatingDirectReservationLink: boolean;
+  createDirectReservationLink: Function;
+};
+
+type RenderTargetProps = {
+  fields: any,
+  targets: Array<Object>,
+  change: Function,
+  formValues: Object,
+};
+
+const renderTargets = ({
+  fields,
+  targets,
+  change,
+  formValues,
+}: RenderTargetProps): Element<*> => <Fragment>
+  {!!targets && targets.map((target) => {
+    const targetIdentifier = target.plan_unit.identifier || target.custom_detailed_plan.identifier;
+
+    const isChecked = (): boolean => {
+      let hit = false;
+      fields.forEach((field, index) => {
+        if(fields.get(index) === target.id) {
+          hit = true;
+        }
+      });
+      return hit;
+    };
+
+    const handleChange = () => {
+      if(isChecked()) {
+        change('targets', formValues['targets'].filter(id => id !== target.id));
+      } else {
+        change('targets', [...formValues['targets'], target.id]);
+      }
+    };
+
+    return <Column lg={4} sm={4} key={target.id}>
+      <FormFieldLabel>Kohteen tunnus</FormFieldLabel>
+      <label className='option-label'>
+        <input
+          type='checkbox'
+          checked={isChecked()}
+          name={`${target.id}-selected`}
+          onChange={handleChange}
+          value={target.id}
+        />
+        <span>{targetIdentifier}</span>
+      </label>
+    </Column>;
```

Comment:
I feel like this could be implemented with the standard `<FormField>` with manually specified attributes, instead of building the input part manually. That might remove the need for the custom styling as well.

Reply:
yup. I did this how the designer has drawed. But I agree :+1:  Let's consult designer with this one and leave it as is?

Reply:
Yeah, let's adjust this later.